test/zhr_devs/submissions/commands/solution_submitted_test.exs

```diff
@@ -0,0 +1,128 @@
+defmodule ZhrDevs.Submissions.Commands.SolutionSubmittedTest do
+  use ExUnit.Case, async: true
+
+  import Commanded.Assertions.EventAssertions
+
+  import ZhrDevs.Fixtures
+
+  alias Commanded.Aggregates.Aggregate
+
+  alias ZhrDevs.App
+
+  alias ZhrDevs.Submissions.{Aggregates, Commands, Events}
+
+  alias ZhrDevs.Submissions.SubmissionIdentity
+
+  describe "SolutionSubmitted command" do
+    setup do
+      valid_opts = fn identity ->
+        [
+          hashed_identity: DomaOAuth.hash(identity),
+          task_uuid: "Jaju6yAv1oZ23NcjJk-1JkOrxCemsH_K-9iRRw0sYRg=",
+          technology: "elixir",
+          solution_path: "test/support/testfile.txt"
+        ]
+      end
+
+      %{valid_opts: valid_opts}
+    end
+
+    test "return error if invalid options passed" do
```

Comment:
These tests are nice to have, but we should make sure that dialyzer catches these errors.

Conversely, for most other things, we need to use proptests and state machine tests.

Again, it's nice to have a unit test as an inspiration for more total testing strategies, but I wouldn't be mad if you instead have demonstrated that you have checked with dialyzer that this sort of thing is impossible.

Reply:
Can dialyzer really prevent us from passing invalid options? I'm really not sure.

Reply:
Yeah, you can type maps with required and optional. I don't think you can type kwlists.